pkg/cluster/sync.go

```diff
@@ -895,8 +895,10 @@ func (c *Cluster) syncConnectionPooler(oldSpec,
 
 	if oldNeedConnectionPooler && !newNeedConnectionPooler {
 		// delete and cleanup resources
-		if err = c.deleteConnectionPooler(); err != nil {
-			c.logger.Warningf("could not remove connection pooler: %v", err)
+		for _, role := range [2]PostgresRole{Master, Replica} {
```

Comment:
This is a bit too ambitious in the sync. What if master is disable and replica is enabled?